spec/tree_spec.rb

```diff
@@ -1,14 +1,54 @@
 require 'rspec'
 require 'tree'
 
-describe 'Tree' do
-  it 'should be a Class' do
-    expect(described_class.is_a? 'Class').to be_true
+describe AppleTree do
+  let(:tree) { AppleTree.new }
+
+  it { expect(tree.class).to be AppleTree  }
+
+  it 'will age' do
+  	tree.age!
+    expect(tree.age).to eq 1
+  end
+
+  it 'can add apples' do
+  	tree.age = 4
+    tree.add_apples
+    expect(tree.any_apples?).to be true
+  end
+
+  context 'when picking apples' do
+
+  	it 'raises an error for no apples' do
+  	  expect { tree.pick_an_apple! }.to raise_error(NoApplesError, "This tree has no apples")
+  	end
+
+  	it 'removes an apple when present' do
+  	  tree.age!
+  	  remaining_apples = tree.apples.length - 1
+  	  tree.pick_an_apple!
+  	  expect(tree.apples.length).to eq remaining_apples
```

Comment:
`remaining_apples` makes it sounds like the value is an array of apples.  Maybe `remaining_apple_count` or similar.